src/components/common/Calculator.js

```diff
@@ -45,6 +48,36 @@ const radToDegree = (rad) => {
   return rad * (180 / Math.PI);
 };
 
+// use special cases because of rounding error
+const getSpecialCase = (angle, fn, isRadian) => {
+  let quarter;
+  let remaining;
+  if (isRadian) {
+    // radian
+    quarter = Math.ceil((angle / (Math.PI / 4)) * 45) % 360;
+    remaining = angle % (Math.PI / 4);
+  } else {
+    // degree
+    quarter = angle % 360;
+    remaining = angle % 45;
+  }
+
+  if (angle < 0) {
+    quarter += 360;
+  }
+
+  // not a special case
+  if (remaining > ROUND_OFF_ERROR_MARGIN) {
```

Comment:
Not sure how this works.

Reply:
Because of round-off errors (computations) even though the value in cos/sin/tan might be 90 but the computation of `remaining` won't give 0. Therefore I need to keep a certain amount of margin to allow a special case.